src/components/Api.js

```diff
@@ -0,0 +1,104 @@
+const url = table => `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${table}`;
+
+const fetchAndSortTodos = async (table, sortOptions = undefined) => {
+  const options = {
+    method: 'GET',
+    headers: {
+      Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_KEY}`,
+    },
+  };
+  const endpoint = sortOptions
+  ? `${url(table)}?${sortOptions}`
+  : url(table);
+  const response = await fetch(endpoint, options);
+  if (!response.ok) {
+    const message = `Error: ${response.status}`;
+    throw new Error(message);
+  }
+  const data = await response.json();
+  const todos = data.records.map((todo) => ({
+    title: todo.fields.title,
+    id: todo.id,
+    createdTime: todo.createdTime,
+    isChecked: todo.done,
+  }));
+
+  const sortedTodos = todos.sort((a, b) => {
+    if (a.createdTime > b.createdTime) {
+      return 1;
+    } if (a.createdTime < b.createdTime) {
+      return -1;
+    }
+    return 0;
```

Comment:
in the case where you're sorting anything numerical, you can use the subtraction cheat to save some effort: 

```suggestion
    return a.createdTime - b.createdTime;
```
I'm sorry I get this wrong 100% of the time, but it's either this or need to swap the `a` and the `b` 😬

Reply:
Somehow, it doesn't work with my code and disregards createdTime. I think it has to do with my API call.

Reply:
Yeah you're right! 

It would only work if we were dealing with the version of a date or time which is a huge number of milliseconds, and here we're dealing with the string-based time.

If it's working successfully with the previous technique you had, that's great (I think that's what I saw when testing).

Reply:
Ok I did some testing and for this way to work it would have to be that each string is converted to a`Date` before they're subtracted, eg: 

```
    if (isSortedAscending) {
      return new Date(objectA.createdTime) - new Date(objectB.createdTime);
    } else {
      return new Date(objectB.createdTime) - new Date(objectA.createdTime);
    }
```